src/app/components/header/header.component.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  ChangeDetectionStrategy, Component, EventEmitter, Output,
+} from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HeaderComponent {
+  searchCards = '';
+
+  @Output() ShowEvent = new EventEmitter<boolean>();
```

Comment:
The same. In general it will work but I it will be better to use Angular style guide https://angular.io/guide/styleguide